### PR TITLE
initial stuff

### PR DESCRIPTION
**Pull request by armw4**
:wink: 

https://github.com/lanetix/elasticsearch-search-stream/blob/initial-branch/README.md

### CODE REVIEW

**Comment by armw4** [comment]
@lanetix/engineers

**Comment by armw4** [comment]
By the time I discovered that https://github.com/hmalphettes/elasticsearch-streams actually allowed you to stream stuff out as opposed to just in, this guy was already born. I haven't tried that guy but it can be an alternative if it comes down to it. I believe this API is a little more friendly; but debatable of course (:wink:).

File: index.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    super({ objectMode: true })
+
+    const { client, search } = options
+
+    this.client = client
+    this.search = search
+
+    this.search.scroll = search.scroll || '30s'
+
+    this._documentsRead = 0
```
Comment (tone81) [inline index.js]: Consider using symbols if you want these properties to be private.

File: index.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    this._scrollId = null
+  }
+
+  _next (error, { hits: { hits, total }, _scroll_id }) {
```
Comment (tone81) [inline index.js]: Consider a symbol to make this function private.

File: test/support/fake-search.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
+const sinon = require('sinon')
```
Comment (tone81) [inline test/support/fake-search.js]: Noticed you were using modules up to this point. Why the change back to commonjs?

**Comment by tone81** [comment]
Had some minor comments; otherwise, :shipit: .

File: index.js (hunk 2 above)
Comment (armw4) [inline index.js]: 👍

File: test/support/fake-search.js (hunk 3 above)
Comment (armw4) [inline test/support/fake-search.js]: Was leftover since I was using only node 4.0 features at first. Converted to es6 after the fact. Will swap this guy.

File: package.json [hunk 4]
```diff
@@ -0,0 +1,42 @@
+{
+  "name": "elasticsearch-search-stream",
+  "version": "0.0.1",
```
Comment (apechimp) [inline package.json]: why not publish as `1.0.0`? We're using it, right?

**Comment by apechimp** [comment]
You'll need a `.npmignore` file so that `src/index.js` isn't ignored. By default, `.gitignore` is used. See [this](https://docs.npmjs.com/misc/developers#keeping-files-out-of-your-package)

File: test/index.js [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+      done()
+    })
+  })
+
+  it('should yield all documents if documents are returned and scrolling is required', (done) => {
+    const options = optionsFactory({ scroll: '1000s' })
+    const searchStream = new SearchStream(options)
+    const fakeSearch = new FakeSearch(searchStream)
+
+    const initialHits = [{ name: 'Danny-Dollarz', lastName: 'T-Breezy' }, { name: 'Professor-Pythonesque', lastName: 'Kyspy-Redskins' }]
```
Comment (apechimp) [inline test/index.js]: I don't think @kyspy would want to be associated with the name of the Washington Football team :stuck_out_tongue:

File: package.json (hunk 4 above)
Comment (armw4) [inline package.json]: I'm noticing this trend where smaller libs never release a major. It seems like frameworks n bigger libs go with a major while other stuff is like 0.7.2 or something. i.e. node-csv, pg-query-stream, and more. Do you know why this is? I'm kool either way but have saw this for some time now.

File: test/index.js (hunk 5 above)
Comment (armw4) [inline test/index.js]: 😳

File: test/index.js [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+    const searchStream = new SearchStream(options)
+    const fakeSearch = new FakeSearch(searchStream)
+
+    const hits = [{ name: 'Danny-Dollarz', lastName: 'Washington' }, { name: 'Professor-Pythonesque', lastName: 'Redskins' }]
+    const documents = []
+
+    options.client.search = fakeSearch.searchSpy({ hits: hits })
+
+    searchStream
+    .pipe(through2.obj((chunk, enc, callback) => {
```
Comment (armw4) [inline test/index.js]: Maybe I can just subscribe to on('data') in place of through2. Didn't consider that

**Comment by armw4** [comment]
👍 for the ignore file.

File: package.json (hunk 4 above)
Comment (apechimp) [inline package.json]: [npm recomends](https://docs.npmjs.com/getting-started/semantic-versioning#semver-for-publishers) to start with `1.0.0`

Comment (armw4) [inline package.json]: Just for Showtime 😉

**Comment by apechimp** [comment]
Outside of those comments,

:sparkles::sparkles::sparkles::sparkles::sparkles::sparkles:
:sparkles::sparkles::sparkles::sparkles::sparkles::sparkles:
:sparkles::sparkles:LGTM:sparkles::sparkles:

File: test/index.js (hunk 6 above)
Comment (apechimp) [inline test/index.js]: Actually, what you probably want here is [`concat-stream`](https://github.com/maxogden/concat-stream)

Comment (armw4) [inline test/index.js]: Aaah. Noticed the object mode guy. Will give that a go.

File: index.js (hunk 2 above)
Comment (armw4) [inline index.js]: Ima pass on the symbols for now. Not a huge deal if u can see a few internals (it's been this way for....everyears.....and.... :wink:).

**Comment by armw4** [comment]
@tone81 @apechimp thanks for teh feedbax n stuffez
